tensorflow/training/docker/2.5/py3/cu112/Dockerfile.gpu

```diff
@@ -0,0 +1,327 @@
+   #libcublas-11-0=11.2.0.252-1 \
+   #libcublas-dev-11-0=11.2.0.252-1 \
+   libcublas-dev-${CUDA_DASH} \
+   libcublas-${CUDA_DASH} \
+   # The 'apt-get install' of nvinfer-runtime-trt-repo-ubuntu1804-5.0.2-ga-cuda10.0
+   # adds a new list which contains libnvinfer library, so it needs another
+   # 'apt-get update' to retrieve that list before it can actually install the
+   # library.
+   # We don't install libnvinfer-dev since we don't need to build against TensorRT,
+   # and libnvinfer4 doesn't contain libnvinfer.a static library.
+   # nvinfer-runtime-trt-repo doesn't have a 1804-cuda10.1 version yet. see:
+   # https://developer.download.nvidia.cn/compute/machine-learning/repos/ubuntu1804/x86_64/
+
+   && rm -rf /var/lib/apt/lists/* \
+   && mkdir -p /var/run/sshd
+
+RUN cd /tmp \
+  && git clone https://github.com/NVIDIA/nccl.git -b v$NCCL_VERSION-1 \
+  && cd nccl \
+  && make -j64 src.build BUILDDIR=/usr/local \
+  && rm -rf /tmp/nccl
+
+# Install EFA without AWS OPEN_MPI
+RUN mkdir /tmp/efa \
+  && cd /tmp/efa \
+  && curl -O https://s3-us-west-2.amazonaws.com/aws-efa-installer/aws-efa-installer-${EFA_VERSION}.tar.gz \
+  && tar -xf aws-efa-installer-${EFA_VERSION}.tar.gz \
+  && cd aws-efa-installer \
+  && ./efa_installer.sh -y --skip-kmod -g \
+  && rm -rf $OPEN_MPI_PATH \
+  && rm -rf /tmp/efa \
+  && rm -rf /tmp/aws-efa-installer-${EFA_VERSION}.tar.gz
+
+# Install OpenMPI without libfabric support
+RUN mkdir /tmp/openmpi && \
+  cd /tmp/openmpi && \
+  wget --quiet https://download.open-mpi.org/release/open-mpi/v4.1/openmpi-${OMPI_VERSION}.tar.gz && \
+  tar zxf openmpi-${OMPI_VERSION}.tar.gz && \
+  cd openmpi-${OMPI_VERSION} && \
+  ./configure --enable-orterun-prefix-by-default --prefix=$OPEN_MPI_PATH && \
+  make -j $(nproc) all && \
+  make install && \
+  ldconfig && \
+  cd / && \
+  rm -rf /tmp/openmpi
+
+RUN wget https://sourceforge.net/projects/boost/files/boost/1.73.0/boost_1_73_0.tar.gz/download -O boost_1_73_0.tar.gz \
+   && tar -xzf boost_1_73_0.tar.gz \
+   && cd boost_1_73_0 \
+   && ./bootstrap.sh \
+   && ./b2 threading=multi --prefix=/usr -j 64 cxxflags=-fPIC cflags=-fPIC install || true \
+   && cd .. \
+   && rm -rf boost_1_73_0.tar.gz \
+   && rm -rf boost_1_73_0 \
+   && cd /usr/include/boost
+
+###########################################################################
+# Horovod & its dependencies
+###########################################################################
+
+# Create a wrapper for OpenMPI to allow running as root by default
+RUN mv $OPEN_MPI_PATH/bin/mpirun $OPEN_MPI_PATH/bin/mpirun.real \
+ && echo '#!/bin/bash' > $OPEN_MPI_PATH/bin/mpirun \
+ && echo 'mpirun.real --allow-run-as-root "$@"' >> $OPEN_MPI_PATH/bin/mpirun \
+ && chmod a+x $OPEN_MPI_PATH/bin/mpirun
+
+# Configure OpenMPI to run good defaults:
+#   --bind-to none --map-by slot --mca btl_tcp_if_exclude lo,docker0
+RUN echo "hwloc_base_binding_policy = none" >> /usr/local/etc/openmpi-mca-params.conf \
```

Comment:
path is not /usr/local anymore. It's moved to openmpi_path [/opt/amazon..]

Refer tf2.4 dockerfile

https://github.com/aws/deep-learning-containers/blob/f9ee2faee183fa87049efb892648714e0d27fe68/tensorflow/training/docker/2.4/py3/cu110/Dockerfile.gpu#L154-L155

This causes failures while benchmarking horovod-based training

Reply:
done